src/python-common/ceph/deployment/service_spec.py

```diff
@@ -2507,6 +2507,12 @@ def __init__(
         # config-key store uri (example:
         # `rados:mon-config-key:smb/config/mycluster/join1.json`).
         join_sources: Optional[List[str]] = None,
+        # user_sources - a list of pseudo-uris that resolve to a (JSON) blob
+        # containing data the samba-container can use to create users (and/or
+        # groups). A ceph based samba container may typically use a rados uri
+        # or a mon config-key store uri (example:
+        # `rados:mon-config-key:smb/config/mycluster/join1.json`).
+        user_sources: Optional[List[str]] = None,
```

Comment:
The non-AD deployment mode has a ceph administrator defining users and groups. These are roughly the same as the `ceph.smb.usersgroups` resource. When we need to deliver this config to the actual smb containers we will store the generated JSON in the priv store and then the URI values for each one will be added to the user_sources list. I suppose it ought to be called `user_group_sources` but I guess I was being lazy. Do you think I ought to change it?

Reply:
no, I don't think the naming was the problem for me here. I think we talked about this offline and you said this was mostly going to be filled in by the SMB module to tell the orchestrator where it stored these credentials so that can be passed on to the SMB daemons. When I first read this I thought the expectation was for users to go and place some auth info at some arbitrary path in the config-key store or some rados object and then just pass that to the orchestrator themselves. But that doesn't seem to be the intended workflow.